DEB-9: Patient immunization

bradsawadye opened the pull request:
DEB-9

bradsawadye committed:
add the channel for the patient immunization

DEB-9

MattyJ007 commented:
@bradsawadye This is not an acceptable Commit message.
See the examples in our ways of [working document](https://docs.google.com/document/d/1fiFr1q4tk4sh6K6Sd7g0hjB8shGMfmY-ZNE3c8qG3kg/edit)

de-laz commented on docker/importer/volume/openhim-import.json:
```diff
@@ -1,52 +1,61 @@
-          "name": "Generic Mapping Mediator",
+          "name": "Hapi Fhir",
+          "secured": false,
           "host": "localhost",
```
@bradsawadye Looks good man. Is there any reason why we are not using the docker compose service name for the FHIR server here? I believe it's `hapi-fhir`, the service name

bradsawadye replied:
@de-laz thanks for the feedback. The issue has been addressed in https://github.com/jembi/covid19-immunization-tracking-package/pull/2/commits/a4db5873c692af2983bd13b3047fbf1a2b7cede2

bradsawadye replied:
@de-laz thanks for the feedback. The issue has been addressed in https://github.com/jembi/covid19-immunization-tracking-package/pull/2/commits/a4db5873c692af2983bd13b3047fbf1a2b7cede2

bradsawadye committed:
correct the route for the OpenHIM patient immunization channel

THe route was incorrect and is now pointing to the right location

DEB-9 DEB-23

bradsawadye committed:
add project details and dependencies used for testing

This adds the package.json and yarn lock files with the dependencies used in the end to end tests

DEB-9

bradsawadye committed:
add environment varibles used in the end to end tests

These varibles enable us to configure the urls and ports for the tests

DEB-9

bradsawadye committed:
add feature file for immunization flow

This is the feature file containing the steps to be run for a successful immunization

DEB-24 DEB-9

bradsawadye committed:
add the feature file for testing channel authorization

THis feature file contains the steps to be executed when running tests for checking the openhim channel authorization. Clients that are not authorized to acces the channel should not be able to access it

DEB-9 DEB-23

bradsawadye committed:
add the step definitions for the immunization flow test

This adds the step definitions for the gherkin steps in the feature files

DEB-9 DEB-23

bradsawadye committed:
add generic functionality for creating reading and deleting fhir resources

This is the functinality that will be used in the gherkin step definitions for sending reading and deleting resources in the tests

DEB-9 DEB-23

bradsawadye committed:
add sample immunzation payload

This will be used in tests.

DEB-9 DEB-23

bradsawadye committed:
add the functionality for sending reading and deleting the immunization resources

These are the methods used in the immunization gherkin step definitions for sending and verifying that resources have been sent successfully

DEB-9 DEB-23

bradsawadye committed:
add openhim patient search channel

THis is the openhim channel for searching patients

DEB-28

bradsawadye committed:
add OpenHIM channels for patient search and immunization

THis is for the kubernetes scripts

DEB-28

MattyJ007 commented on features/immunization/utils.js:
```diff
@@ -0,0 +1,40 @@
+let authorizationError
+
+const immunization = JSON.parse(
+  fs.readFileSync(`${__dirname}/payloads/immunization.json`, 'utf8')
```
@bradsawadye This directory doesn't exist

MattyJ007 commented on package.json:
```diff
@@ -0,0 +1,32 @@
+  "version": "0.0.1",
+  "main": "index.js",
+  "scripts": {
+    "lint": "eslint ./features/*/*.js",
+    "lint:fix": "eslint ./features/*/*.js --fix",
```
These two commands don't work

MattyJ007 reviewed (changes requested):
@bradsawadye There are broken commands in these scripts

MattyJ007 commented on package.json:
```diff
@@ -0,0 +1,32 @@
+    "lint": "eslint ./features/*/*.js",
+    "lint:fix": "eslint ./features/*/*.js --fix",
+    "test": "env-cmd -f ./docker/.env.local cucumber-js -f progress-bar ./features/",
+    "test:remote": "env-cmd -f ./ansible/.env.remote cucumber-js -f progress-bar ./features/"
```
@bradsawadye This is not an ansible project

MattyJ007 commented on docker/importer/volume/openhim-import.json:
```diff
@@ -1,52 +1,118 @@
-      "endpoints": [
+      "methods": [
+        "GET",
+        "POST",
+        "DELETE",
+        "PUT"
```
@bradsawadye This is a search channel - only GET

MattyJ007 replied:
@bradsawadye This is a search channel only - only GET

MattyJ007 reviewed (changes requested):
@bradsawadye 
Looking good for the patient search channel just one small change 👍🏼

bradsawadye committed:
remove the methods not used on channel

THe patient search channel is for searching only and only the get method is used

DEB-28

bradsawadye committed:
fix typo

the name of the folder holding the immunization payload is 'payload' not 'payloads'

DEB-24

MattyJ007 committed:
Merge branch 'master' into DEB-9-patient-vaccination

MattyJ007 committed:
Update README with testing instructions

to make start up and testing easier

DEB-8

MattyJ007 commented on README.md:
```diff
@@ -64,3 +71,8 @@ TODO: Channel config.
+You should receive the created form of the Patient resource in HAPI FHIR - notice the Resource now has an `id` field.
+
+To **update** a patient, send through the payload above to `/patient-registration/{FHIR_ID}` but add the field `id` to the object root.
+Also substitute in the `id` into the Request path in the `FHIR_ID` placeholder position.
```
@bradsawadye Could you please add some examples of the Patient search feature here as well as the immunization

MattyJ007 reviewed (approved):
@bradsawadye I'm approving this PR so that my `requested changes` don't block the PR whilst I'm on leave.
But there are still a few changes that are needed -
@de-laz could you please test my change requests when they're implemented

bradsawadye committed:
set the immunizationId

this is so the id is global and can be used in al mehtods in the module

DEB-31

bradsawadye committed:
make the http method configurable

THis is sos the method for sending request is generic

DEB-31

bradsawadye committed:
add the paatient payload

This is the patient payload that will be used in the immunization test flow

DEB-24

bradsawadye committed:
add functionality for testing the petient search

This adds the feature files, step definitions and functionality used in the end to end tests for the patient search flow

DEB-29

bradsawadye committed:
add valid patient resource

The resource was invalid

DEB-29

bradsawadye committed:
modify method for sending immunizaton

The result of the operation is an empty object and the id of the resouce created can not be retrieved from the response body

DEB-24

bradsawadye committed:
verify method for retrieving resources

The id of the resource created is not accessible so the a different search parameter is now being used

DEB-29

bradsawadye committed:
modify the methods

The method for getting the patient has been modified to use the name parameter instead of the id which is not easily accessible

DEB-29

bradsawadye committed:
use a unique reference for the immunization

DEB-29

bradsawadye committed:
fix functionality for getting patient

The result of a search is a fhir search bundle. The patient is nested in the bundle

DEB-29

bradsawadye committed:
Merge branch 'DEB-9-patient-vaccination' of github.com:jembi/covid19-immunization-tracking-package into DEB-9-patient-vaccination

bradsawadye committed:
remove unused variable

DEB-29

bradsawadye committed:
add feature files payloads and step definitions

THis adds the feature files, payloads, and the functionality used in testing the registration flow (end to end)

DEB-31

bradsawadye committed:
add example payload for immunization

This is a sample payload that can be used for testing

DEB-24

bradsawadye committed:
correct the path for testing remote deployments

THe environment variables required for testing remote deployments are contained in the kubernetes folder

DEB-31

bradsawadye committed:
fix linting issues

DEB-31

bradsawadye committed:
fix more lint issues

DEB-29

de-laz commented:
@bradsawadye As discussed, I will be merging this branch and the remaining work will be done on a new branch.
